akd/src/append_only_zks.rs

```diff
@@ -637,47 +638,132 @@ impl Azks {
             })
             .collect();
 
-        // Load nodes.
-        self.preload_nodes(storage, &AzksElementSet::from(lookup_nodes))
+        // Load nodes without parallelism, since multiple lookups could be
+        // happening and parallelism might consume too many resources.
+        self.preload_nodes(storage, &AzksElementSet::from(lookup_nodes), true)
```

Comment:
nit: Shoulda-been-an-enum :p

Reply:
We talked about this, and personally I'm inclined to introduce the indirection if the use of the param is slightly more complex than what we have here (either having multiple variants or being used externally). But given that I've heard this feedback twice, I'll adopt it :)